worker/src/ee/cloudSpendAlerts/handleCloudSpendAlertJob.ts

```diff
@@ -0,0 +1,258 @@
+    );
+    traceException(
+      `[CLOUD SPEND ALERTS] Stripe customer id not found for org ${orgId}`,
+    );
+    return;
+  }
+  // Get Stripe subscription ID
+  const stripeSubscriptionId = org.cloudConfig?.stripe?.activeSubscriptionId;
+  if (!stripeSubscriptionId) {
+    logger.error(
+      `[CLOUD SPEND ALERTS] Stripe subscription id not found for org ${orgId}`,
+    );
+    traceException(
+      `[CLOUD SPEND ALERTS] Stripe subscription id not found for org ${orgId}`,
+    );
+    return;
+  }
+
+  // Setup stripe client
+  const stripe = new Stripe(env.STRIPE_SECRET_KEY);
+
+  try {
+    // Get subscription to check billing cycle
+    const subscription =
+      await stripe.subscriptions.retrieve(stripeSubscriptionId);
+
+    const currentPeriodStart = new Date(
+      subscription.current_period_start * 1000,
+    );
+
+    // Create preview invoice to calculate current spend
+    const canCreateInvoicePreview = [
+      "active",
+      "past_due",
+      "trialing",
+      "unpaid",
+    ].includes(subscription.status);
+
+    const previewInvoice = canCreateInvoicePreview
+      ? await backOff(
+          async () =>
+            await stripe.invoices.createPreview({
+              customer:
+                typeof subscription.customer === "string"
+                  ? subscription.customer
+                  : subscription.customer?.id,
+              subscription: stripeSubscriptionId,
+            }),
+          {
+            numOfAttempts: 3,
+          },
+        )
+      : null;
+
+    // Calculate current spend in USD
+    const currentSpendCents = previewInvoice?.total ?? 0;
+    const currentSpendUSD = currentSpendCents / 100;
+
+    logger.info(
+      `[CLOUD SPEND ALERTS] Org ${orgId} current spend: $${currentSpendUSD.toFixed(2)}`,
+    );
+
+    let countTriggeredAlerts = 0;
+
+    // Check each spend alert for this org
+    for (const alert of org.spendAlerts) {
+      const thresholdUSD = parseFloat(alert.threshold.toString());
+
+      // Check if threshold is breached
+      if (currentSpendUSD >= thresholdUSD) {
+        // Check if already triggered this billing cycle
+        const alreadyTriggered =
+          alert.triggeredAt && alert.triggeredAt >= currentPeriodStart;
+
+        if (!alreadyTriggered) {
+          logger.info(
+            `[CLOUD SPEND ALERTS] Triggering alert ${alert.id} for org ${orgId} - spend $${currentSpendUSD.toFixed(2)} >= threshold $${thresholdUSD.toFixed(2)}`,
+          );
+
+          // Get org admins and owners for email notifications
+          const adminMemberships = await prisma.organizationMembership.findMany(
+            {
+              where: {
+                orgId: orgId,
+                role: { in: [Role.OWNER, Role.ADMIN] },
+              },
+              include: {
+                user: {
+                  select: { email: true },
+                },
+              },
+            },
+          );
```

Comment:
We could move this out of the loop or cache the result. If there are 100 triggered alerts for an org, we'll fetch the users 100 times.

Reply:
ok